bundles/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/internal/MyStromAbstractHandler.java

```diff
@@ -0,0 +1,173 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mystrom.internal;
+
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_CONNECTED;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_DNS;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_GW;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_IP;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_LAST_REFRESH;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_MAC;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_MASK;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_SSID;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_STATIC;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_TYPE;
+import static org.openhab.binding.mystrom.internal.MyStromBindingConstants.PROPERTY_VERSION;
+
+import java.text.DateFormat;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link MyStromAbstractHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Frederic Chastagnol - Initial contribution
+ */
+@NonNullByDefault
+public abstract class MyStromAbstractHandler extends BaseThingHandler {
+    protected static final int HTTP_OK_CODE = 200;
+    protected static final String COMMUNICATION_ERROR = "Error while communicating to the myStrom plug: ";
+    protected static final String HTTP_REQUEST_URL_PREFIX = "http://";
+
+    private final HttpClient httpClient;
+    private String hostname = "";
+    private String mac = "";
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private final Gson gson = new Gson();
+
+    public MyStromAbstractHandler(Thing thing, HttpClient httpClient) {
```

Comment:
We usually add `Abstract` at the beginning, so `AbstractMyStromHandler` would be a better naming.